tools/AutoTuner/src/autotuner/utils.py

```diff
@@ -71,6 +70,56 @@
 DATE = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
 
 
+# --- General utilities ---
+def run_command(
+    args, cmd, timeout=None, stderr_file=None, stdout_file=None, fail_fast=False
+):
+    """
+    Wrapper for subprocess.run
+    Allows to run shell command, control print and exceptions.
+    """
+    process = subprocess.run(
+        cmd, timeout=timeout, capture_output=True, text=True, check=False, shell=True
+    )
+    if stderr_file is not None and process.stderr != "":
+        with open(stderr_file, "a") as file:
+            file.write(f"\n\n{cmd}\n{process.stderr}")
+    if stdout_file is not None and process.stdout != "":
+        with open(stdout_file, "a") as file:
+            file.write(f"\n\n{cmd}\n{process.stdout}")
+    if args.verbose >= 1:
+        print(process.stderr)
+    if args.verbose >= 2:
+        print(process.stdout)
+
+    if fail_fast and process.returncode != 0:
+        raise RuntimeError
+
+
+def set_seed(seed: int):
+    """Set seed for reproducibility."""
+    import torch
+    import random
```

Comment:
Adding the import inside a `def` leads to late dependency resolution. I would prefer to fail fast in this case.